src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs

```diff
@@ -177,18 +185,26 @@ private void AddStyle_SaveButton_Click(object sender, RoutedEventArgs e)
                 newItem.Name = "Input";
 
             //if the validation returns false it means that the new style that will be added doesn't exists
-            if (viewModel.ValidateExistingStyle(newItem) == false)
+            if (string.IsNullOrEmpty(groupNameLabel.Text))
             {
-                viewModel.AddStyle(newItem);
-                viewModel.ResetAddStyleControl();
-                AddStyleBorder.Visibility = Visibility.Collapsed;
-                AddStyleButton.IsEnabled = true;
+                viewModel.CurrentWarningMessage = Res.PreferencesViewEmptyStyleWarning;
+                viewModel.IsWarningEnabled = true;
+                viewModel.IsSaveButtonEnabled = false;
             }
-            else
+            else if (viewModel.ValidateExistingStyle(newItem) == true)
             {
+                viewModel.CurrentWarningMessage = Res.PreferencesViewAlreadyExistingStyleWarning;
                 viewModel.IsWarningEnabled = true;
+                viewModel.IsSaveButtonEnabled = false;
             }
-            
+            else
+            {
+                viewModel.AddStyle(newItem);
+                viewModel.ResetAddStyleControl();
+                AddStyleBorder.Visibility = Visibility.Collapsed;
+                AddStyleButton.IsEnabled = true;
+                viewModel.IsSaveButtonEnabled = true;
```

Comment:
This is already done inside ResetAddStyleControl(), right?

Reply:
@reddyashish ResetAddStyleControl() method is doing other stuff.
![image](https://user-images.githubusercontent.com/61755417/158252473-7de7304b-2332-48b1-a899-3b41a426594f.png)

Reply:
@RobertGlobant20 Then should we add these into the reset function as well?

Reply:
Oh yeah I got confused with the ResetGroupStyleForm().

Reply:
@QilongTang I added those lines in the ResetGroupStyleControl() method and also I refactored some code, please check the next commit: https://github.com/DynamoDS/Dynamo/pull/12691/commits/66758055b162acac6f5bdf7d110062eb2c55479a